webapp/src/adapters/solid/PODManager.ts

```diff
@@ -0,0 +1,164 @@
+import { QueryEngine } from '@comunica/query-sparql-solid';
+import { saveSolidDatasetAt, SolidDataset } from '@inrupt/solid-client';
+import Map from '../../domain/Map';
+import Assembler from './Assembler';
+import SolidSessionManager from './SolidSessionManager';
+import Placemark from '../../domain/Placemark';
+import Place from '../../domain/Place';
+
+export default class PODManager {
+    private sessionManager: SolidSessionManager  = SolidSessionManager.getManager();
+
+
+    public async savePlace(place:Place): Promise<boolean> {
+        let path:string = this.getBaseUrl() + '/data/places/' + place.uuid;
+
+        return this.saveDataset(path, Assembler.placeToDataset(place))
+            .then(() => {return true})
+            .catch(() => {return false});
+    }
+
+    /**
+     * Saves a map to the user's POD
+     * 
+     * @param map the map to be saved
+     * @returns wether the map could be saved
+     */
+    public async saveMap(map:Map): Promise<boolean> {
+        let path:string = this.getBaseUrl() + '/data/maps/' + map.getId();
+
+        return this.saveDataset(path, Assembler.mapToDataset(map))
+            .then(() => {return true})
+            .catch(() => {return false});
+    }
+
+    public async loadPlacemarks(map: Map): Promise<void> {
+        let path:string = this.getBaseUrl() + '/data/maps/' + map.getId();
+        let placemarks = await this.getPlacemarks(path);
+        map.setPlacemarks(placemarks);
+    }
+
+    /**
+     * Returns the details of all the maps of the user.
+     * The placemarks will not be loaded.
+     * 
+     * @returns an array of maps containing the details to be displayed as a preview
+     */
+    public async getAllMaps(): Promise<Array<Map>> {
+        let path:string = this.getBaseUrl() + '/data/maps/';
+
+        let urls = await this.getContainedUrls(path);
+        let maps = await this.getMapPreviews(urls);
+        return maps;
+    }
+
+    public async getPlace(url:string): Promise<Place> {
```

Comment:
A place also have a category and images

Reply:
It didn't when i created the branch and that issue is assigned to someone else. I just store and retrieve the existing entities. When that part is implemented i will update the POD manager.